content/tutorials/trainapp-fitbit.textile

```diff
@@ -0,0 +1,550 @@
+
+"See this step in Github":https://github.com/ably/tutorials/commit/39c186e06e75008aa77ec5fb2060178b2e32a815
+
+h2(#text-watch). Step 9 - Updating the text on the watch with dynamic values
+
+Now when we get the data that we need using the @messaging@ module to the @app@ from the @companion@.
+Using those values we decide the color of the text eg: Red when time is less . We also set the time to be displayed on the watch.
+Empty the file @index.js@ which is inside the @app@ folder and add the following code:
+
+```[javascript]
+import document from "document";
+// Import the messaging module
+import * as messaging from "messaging";
+let spinner = document.getElementById("spinner");
+
+// Start the spinner
+spinner.state = "enabled";
+
+import clock from "clock";
+
+let myClock = document.getElementById("myClock");
+
+clock.granularity = 'seconds'; // seconds, minutes, hours
+
+// Updating the time on the clockface
+clock.ontick = function(evt) {
+  myClock.text = ("0" + evt.date.getHours()).slice(-2) + ":" +
+                      ("0" + evt.date.getMinutes()).slice(-2) + ":" +
+                      ("0" + evt.date.getSeconds()).slice(-2);
+};
+
+// Get the elements & set their values & colors
+let eta = document.getElementById("eta");
+let teta = document.getElementById("teta");
+let weta = document.getElementById("weta");
+let ll = document.getElementById("leavelabel");
+let tl = document.getElementById("trainlabel");
+let wl = document.getElementById("walkinglabel");
+let cir = document.getElementById("cir");
+messaging.peerSocket.onmessage = function(evt) {
+  // Stop the spinner
+  spinner.state = "disabled";
+  // Adding the text & updating the color of text and arc
+  eta.text = evt.data.lT + "min" ;
+  ll.text = "Leave in";
+  if (evt.data.lT <=3 ) {
+    eta.style.fill = "fb-red"
+    cir.style.fill = "fb-red"
+  } else if (evt.data.lT >3 && evt.data.lT<=5 ) {
+    eta.style.fill = "fb-yellow"
+    cir.style.fill = "fb-yellow" 
+  }
+  teta.text = evt.data.tTA + "min" ;
+  tl.text = "Train in";
+  weta.text = evt.data.wT + "min" ;
+  wl.text = "Walk time";
+}
+```
+
+"See this step in Github":https://github.com/ably/tutorials/commit/6e5b6888d787675fec45e85bfcf6c3379332354e
+
+h2(#permissions-package). Step 9 - Add permissions in package.json
+
+Now lets add the needed permissions to @package.json@. Open the file @package.json@ which is in the base folder and update the field @requestedPermissions@ with the following:
+
+```[javascript]
+[
+  "access_internet",
+  "access_location",
+  "run_background"
+]
+```
+
+"See this step in Github":https://github.com/ably/tutorials/commit/03db8411afc271e12daaff2bdbdc870a16f9133b
+
+So this completes the work of the app and we have a working clockface!
+Go ahead and open the simulator, and run the app using the commands which we used before.
```

Comment:
Newline needed.

Reply:
Perhaps a link or direct mention of what the specific `commands we used before` are.